utilities/hooks/useForm.ts

```diff
@@ -0,0 +1,40 @@
+import { ChangeEventHandler, useReducer } from 'react'
+import { FormProps } from '../types/formTypes'
+interface ActionState {
+    type: string
+    payload?: any
+}
+function reducer(state: FormProps, action: ActionState) {
```

Comment:
Few things here. First off, this looks good overall.

1. Let's start putting line breaks/spaces between things like the `imports` and the next set of code (in this case, it's the `interface`. Line between interface and reducer...so on and so forth
2. Pre-define your types. Never...never want to re-type things in multiple spots. If you have something typed out more than once, it should be a variable.

```
const RESET = 'reset"
const UPDATE = 'update'

...

case RESET: {
...
}

...

dispatch({
  type: RESET,
  payload: {...}
})
```